vaadin-cdi/src/test/java/com/vaadin/cdi/MultipleAccessIsolationTest.java

```diff
@@ -41,17 +41,19 @@ public void testConcurrentAccess() throws IOException,
         assertThat(getCount(ConcurrentUI.CONSTRUCT_COUNT), is(1));
         firstWindow.findElement(By.id(ConcurrentUI.OPEN_WINDOW)).click();
         waitForNumberOfWindowsToEqual(2);
-        Thread.sleep(500);
         List<String> handles = new ArrayList<String>(
                 firstWindow.getWindowHandles());
+        // wait for second window fully loads
+        firstWindow.switchTo().window(handles.get(1));
+        (new WebDriverWait(firstWindow, 15)).until(ExpectedConditions
+                .presenceOfElementLocated(LABEL));
+
         firstWindow.switchTo().window(handles.get(0));
```

Comment:
(No wait needed after any of the other window.switchTo() calls? Is everything already loaded in them? I assume so, haven't dug deeper.)